New error pages

gustavwilliam opened the pull request:
This PR adds pages for 404 and 500 errors on the site. They are following the same structure as the recently merged custom CF error pages (python-discord/error-pages#1).

## Preview
### 404
<img width="1409" alt="Screenshot 2021-03-01 at 17 16 07" src="https://user-images.githubusercontent.com/65498475/109525445-d23dd600-7ab1-11eb-9769-4b1acce7f200.png">

### 500
<img width="1409" alt="Screenshot 2021-03-01 at 17 17 33" src="https://user-images.githubusercontent.com/65498475/109525645-0618fb80-7ab2-11eb-94d3-10ce551139ff.png">

gustavwilliam committed:
Add 404 page

This does not include the styling required for the page to display
properly

gustavwilliam committed:
Add error page styling

This css sheet applies to the 404 and 500 pages

gustavwilliam committed:
Add 500 page

lemonsaurus commented on pydis_site/templates/404.html:
```diff
@@ -0,0 +1,34 @@
+                alt="Python Discord banner" />
+        </div>
+        <div class="content-box">
+            <h1>404 — not found</h1>
```
I'm nitpicking, but in [RFC7231](https://tools.ietf.org/html/rfc7231#section-6.5.4) where this error code is defined, it's styled with capital letters, like 404 Not Found.

```suggestion
            <h1>404 — Not Found</h1>
```

lemonsaurus commented on pydis_site/templates/500.html:
```diff
@@ -0,0 +1,29 @@
+                alt="Python Discord banner" />
+        </div>
+        <div class="content-box">
+            <h1>500 — internal server error</h1>
```
Same thing here, see https://tools.ietf.org/html/rfc7231#section-6.6.1 

```suggestion
            <h1>500 — Internal Server Error</h1>
```

lemonsaurus reviewed (approved):
Looks very good overall. Excellent work, @gustavwilliam.

jb3 commented on pydis_site/templates/500.html:
```diff
@@ -0,0 +1,29 @@
+        </div>
+        <div class="content-box">
+            <h1>500 — internal server error</h1>
+            <p>Sorry, but something went wrong on our side of things.</p>
```
```suggestion
            <p>Something went wrong at our end. Please try again shortly, or if the problem persists please let us know <a href="https://discord.gg/python">on Discord</a></p>
```

jb3 reviewed (changes requested):
Looks good. Minor comment on wording + including a Discord link on the 500 page as well as the 404 one.

gustavwilliam committed:
Capitalise error codes in 404 and 500 pages

Co-authored-by: Leon Sandøy <leon.haland@gmail.com>

gustavwilliam committed:
Update error message for 500 page

Now includes a link to our Discord server and concrete actions for the user, if the problem persists

jb3 reviewed (approved):
:shipit::shipit::shipit::shipit::shipit:
